### DISCUSSION TITLE
Build Eclipse Embedded C/C++ project from command prompt

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-cdt/cdt#899] [General]
Using Eclipse UI, I can build the project successfully. 
However, using command prompt, it failed with WARNING: No Project matched "FX3Project". Skipping...
eclipsec.exe -noSplash -data "D:\Development" -application org.eclipse.cdt.managedbuilder.core.headlessbuild -cleanBuild FX3Project

**Comment by Shriharsh-Deshmukh** [comment]
The issue might be due to workspace/project mismatch or incorrect project path. Ensure the project exists in the specified workspace (`D:\Development`). You can try refreshing the workspace and double-check the project name spelling.

Also, try using:

```bash
eclipsec.exe -noSplash -data "D:\Development" -application org.eclipse.cdt.managedbuilder.core.headlessbuild -cleanBuild all
```

This will clean and build all projects in the workspace.

**Comment by zhiyong-ft** [comment]
Just want to drop a line here that the same can be done on Linux with slight modifications:
```
eclipse -noSplash -data ~/eclipse-workspace -application org.eclipse.cdt.managedbuilder.core.headlessbuild -cleanBuild my-project
```
Replace `~/eclipse-workspace` with actual path to your workspace without the double quote that are needed on Windows, and replace `my-project` with either `all` or actual name of your project.
It took me a while and a bunch of google searches go get it to work. Hopefully it will save some poor souls some time.
